core/src/test/java/org/elasticsearch/deps/joda/SimpleJodaTests.java

```diff
@@ -262,6 +262,26 @@ public void testThatEpochsCanBeParsed() {
         }
     }
 
+    public void testThatFloatEpochsCanBeParsed() {
+
+        long millisFromEpoch = randomNonNegativeLong();
```

Comment:
Since dates previous to epoch 0 can still be expressed as a long and may well be encountered if the user is indexing historical data should we test negative epoch values here too?